chrome/content/zotero/xpcom/data/item.js

```diff
@@ -3751,7 +3754,14 @@ Zotero.Item.prototype.fromJSON = function (json) {
 		case 'dateAdded':
 		case 'dateModified':
 			if (val) {
-				let d = Zotero.Date.isoToDate(val);
+				let d = new Date(val);
+				if (isNaN(d.getTime())) {
+					d = Zotero.Date.sqlToDate(val, true);
+				}
+				if (isNaN(d.getTime())) {
+					d = Zotero.Date.strToDate(val);
+					d = new Date(d.year, d.month, d.day);
+				}
```

Comment:
The data being parsed is supposed to follow a fairly strict standard. When do we have non-ISO dates being passed to this function?